File: RepoTransBench/source_projects/Python/willthames_ansible-inventory-grapher/public_tests/test_public_vault.py
import pytest

# This public test emulates test logic from tests/test_vault.py
# Since the original test likely interacts with ansible vault features,
# we focus on changing filenames, ids, or passphrases, while preserving test intent.

import os

def test_public_fake_vault_file(tmp_path):
    # Create a public-named vault-like file
    vault_file = tmp_path / "public_vaultfile"
    vault_file.write_text("$ANSIBLE_VAULT;1.1;AES256\ntestpublic")
    assert vault_file.exists()
    # Pretend to read/parse it
    with open(vault_file, "r") as f:
        content = f.read()
    assert "$ANSIBLE_VAULT" in content
    assert "public" in content

def test_public_vault_password(tmp_path):
    # Simulate reading a different vault password
    pw_file = tmp_path / "public_vaultpass"
    vault_password = "superpublicpw"
    pw_file.write_text(vault_password)
    assert pw_file.read_text() == vault_password

@pytest.mark.parametrize("vault_content", [
    "$ANSIBLE_VAULT;1.2;AES256\npublicvaultcipher",
    "$ANSIBLE_VAULT;1.2;AES256\nanotherpubliccipher"
])
def test_public_multiple_vault_files(tmp_path, vault_content):
    vault_file = tmp_path / "vault_varied.public"
    vault_file.write_text(vault_content)
    assert "AES256" in vault_file.read_text()

File: RepoTransBench/target_projects/Python/Java/willthames_ansible-inventory-grapher/src/test/java/com/example/ansibleinventorygrapher/publictests/TestPublicVault.java
package com.example.ansibleinventorygrapher.publictests;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.*;
import java.nio.file.*;
import java.util.*;

class TestPublicVault {

    @Test
    void testPublicFakeVaultFile() throws IOException {
        Path tempDir = Files.createTempDirectory("testPublicFakeVaultFile");
        Path vaultFile = tempDir.resolve("public_vaultfile");
        String content = "$ANSIBLE_VAULT;1.1;AES256\ntestpublic";
        Files.writeString(vaultFile, content);

        assertTrue(Files.exists(vaultFile));

        String readContent = Files.readString(vaultFile);
        assertTrue(readContent.contains("$ANSIBLE_VAULT"));
        assertTrue(readContent.contains("public"));
    }

    @Test
    void testPublicVaultPassword() throws IOException {
        Path tempDir = Files.createTempDirectory("testPublicVaultPassword");
        Path pwFile = tempDir.resolve("public_vaultpass");
        String vaultPassword = "superpublicpw";
        Files.writeString(pwFile, vaultPassword);

        String read = Files.readString(pwFile);
        assertEquals(vaultPassword, read);
    }

    @Test
    void testPublicMultipleVaultFiles() throws IOException {
        Path tempDir = Files.createTempDirectory("testPublicMultipleVaultFiles");
        String[] vaultContents = {
            "$ANSIBLE_VAULT;1.2;AES256\npublicvaultcipher",
            "$ANSIBLE_VAULT;1.2;AES256\nanotherpubliccipher"
        };
        for (String vaultContent : vaultContents) {
            Path vaultFile = tempDir.resolve("vault_varied.public");
            Files.writeString(vaultFile, vaultContent);
            String read = Files.readString(vaultFile);
            assertTrue(read.contains("AES256"));
        }
    }
}